app/models/user.rb

```diff
@@ -0,0 +1,34 @@
+require_relative '../data_mapper_setup'
+require 'bcrypt'
+require 'dm-validations'
+
+class User
+  include DataMapper::Resource
+  attr_reader :password
+  attr_accessor :password_confirmation
+  validates_confirmation_of :password
+
+
+  property :id, Serial
+  property :email, String, required: true, :format => :email_address, :unique => true
+  property :password_digest, Text
+  property :name, String
+  property :username, String, required: true, :unique => true
+
+  has n, :peeps
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
+    user = first(email: email)
+    if user && BCrypt::Password.new(user.password_digest) == password
```

Comment:
This feels like the wrong way round. Shouldn't this be:
`BCrypt::Password.new(password) == user.password_digest`
Normally password encryption is one way. I would expect that we would encrypt the password the user supplies and compare that to the digest.